src/isar/apis/schedule/start_mission.py

```diff
@@ -65,8 +65,9 @@ def get(self):
             self.logger.error(message)
             return message, HTTPStatus.NOT_FOUND
 
-        mission: Mission = self.mission_reader.get_mission_by_id(mission_id)
-        if mission is None:
+        try:
+            mission: Mission = self.mission_reader.get_mission_by_id(mission_id)
+        except MissionReaderError:
             message = StartMissionMessages.mission_not_found()
             return message, HTTPStatus.NOT_FOUND
```

Comment:
Since this is at the top level of the functionality I would log the error here. So `except MissionReaderError as e` and `self.logger.error(f"Failed to read mission\n {e}")`